common/browser_process.go

```diff
@@ -58,12 +59,12 @@ func NewBrowserProcess(
 	ctx context.Context, path string, args, env []string, dataDir *storage.Dir,
 	ctxCancel context.CancelFunc, logger *log.Logger,
 ) (*BrowserProcess, error) {
-	cmd, stdout, procDone, err := execute(ctx, path, args, env, dataDir, logger)
+	cmd, procDone, err := execute(ctx, path, args, env, dataDir, logger)
 	if err != nil {
 		return nil, err
 	}
 
-	wsURL, err := parseDevToolsURL(ctx, stdout)
+	wsURL, err := getDevToolsURL(dataDir.Dir)
```

Comment:
If the `getDevToolsURL` was called prior to calling `execute` (which will require some goroutine orchestration so that execute is only called once `getDevToolsURL` has started waiting for the file)  then we should avoid any possible race conditions, no?

Reply:
True, that could be a solution as well. Though a) we're not 100% sure this is caused by a race, see the discussion above, and b) starting a goroutine with proper orchestration and parsing stdout is arguably messier than just reading a file.